presto-main/src/main/java/com/facebook/presto/execution/QueryQueueManager.java

```diff
@@ -26,5 +28,9 @@
 @ThreadSafe
 public interface QueryQueueManager
 {
-    void submit(Statement statement, QueryExecution queryExecution, Executor executor);
+    void submit(Statement statement, QueryExecution queryExecution, SelectionContext<?> group, Executor executor);
+
+    // TODO: Figure out what's the right thing to do for the following interface (e.g. keep it here? merge it into ResourceGroupManager
+    // and deprecate this interface?)
+    SelectionContext<?> selectGroup(SelectionCriteria criteria);
```

Comment:
Once we remove the legacy queueing implementation I think there is no point in keeping this interface. We can have this public interface in `ResourceGroupManager`.